### DISCUSSION TITLE
Pass variable to a directive expression

### DISCUSSION THREAD

**Discussion by Luddinus** [alpinejs/alpine#3863] [1. Help] [answered]
Hello,

I have a button that triggers a modal with a directive like this:

```
<button x-modal:some-modal-id="console.log($modal)">
```

The directive
```
Alpine.directive('modal', (el, { value, modifiers, expression }, { evaluate, evaluateLater })) => {
   const $modal = { foo: 'bar' };

   evaluate(expression);
})
```

How can I pass the $modal variable to the expression?

**Comment by Luddinus** [comment]
Posible solution:

```
Alpine.store('modal', () => {
   current: null
})

Alpine.magic('modal', () => Alpine.store('modal').current)
```

And then, in the directive set the current modal...

**Comment by ekwoka** [comment]
`evaluate` has a second argument, which is an object of global you want to expose when evaluating the script

So `evaluate(expression, { $modal })`

**Reply by Luddinus** [reply]
@ekwoka I though that but I get that "$modal" is undefined...

**Reply by ekwoka** [reply]
Show code. And where does the error come from?

**Reply by Luddinus** [reply]
html
```
<button
      x-data
      class="btn btn-link"
      x-modal:modal-user-list="console.log($modal)"
  >{{ $user->name }}</button>
```

alpine directive
```
Alpine.directive('modal', (el, { value, modifiers, expression }, { evaluate, evaluateLater }) => {
    let clickHandler = (e) => {
        if (expression) {
            evaluate(expression, { $modal: 'foo' })
        }
    }

    el.addEventListener('click', clickHandler)
})
```

**Reply by ekwoka** [reply]
This should definitely work

**Reply by Luddinus** [reply]
Here is the codepen with the error, maybe I'm missing something
https://codepen.io/luddinus/pen/NWowWEe?editors=1011

**Reply by ekwoka** [reply, marked as answer]
Whoops, messed up the type.

The object passed as the second argument needs to be `{ scope: { YOUR STUFF HERE } }`

Obligatory: TypeScript would have solved this :)

**Reply by Luddinus** [reply]
Thanks!
